op-challenger/runner/factory.go

```diff
@@ -38,6 +38,15 @@ func createTraceProvider(
 		}
 		prestateProvider := vm.NewPrestateProvider(prestate, stateConverter)
 		return cannon.NewTraceProvider(logger, m, cfg.Cannon, serverExecutor, prestateProvider, prestate, localInputs, dir, 42), nil
+	case types.TraceTypeCannonKona:
+		serverExecutor := vm.NewKonaExecutor()
+		stateConverter := cannon.NewStateConverter(cfg.Cannon)
+		prestate, err := prestateSource.getPrestate(ctx, logger, cfg.CannonAbsolutePreStateBaseURL, cfg.CannonAbsolutePreState, dir, stateConverter)
```

Comment:
I think this should wind up as:
```suggestion
		stateConverter := cannon.NewStateConverter(cfg.CannonKona)
		prestate, err := prestateSource.getPrestate(ctx, logger, cfg.CannonKonaAbsolutePreStateBaseURL, cfg.CannonKonaAbsolutePreState, dir, stateConverter)
```

And similar changes in other places where CannonKona is accessing the `cfg.Cannon` vm.Config.

Reply:
Done [here](https://github.com/ethereum-optimism/optimism/pull/17249/commits/0b509db2cb58dc5c69f663453a0bbb07ab970c33) 👍